api/controllers/game/rematch.js

```diff
@@ -21,7 +21,14 @@ module.exports = async function (req, res) {
 
     // Determine whether to start new game
     const oldPNum = game.p0 === userId ? 0 : 1;
-    const gameUpdates = { [`p${oldPNum}Rematch`]: rematch };
+    const gameUpdates = {
+      [`p${oldPNum}Rematch`]: rematch,
+      lastEvent: {
+        change: 'rematch',
+        game: { ...game.lastEvent.game, [`p${oldPNum}Rematch`]: rematch },
```

Comment:
I'm wondering if we can DRY up the rematch expression by capturing it in a variable like
```suggestion
    const rematchVal = { [`p${oldPNum}Rematch`]: rematch };
    const gameUpdates = {
      ...rematchVal,
      lastEvent: {
        change: 'rematch',
        game: { ...game.lastEvent.game, ...rematchVal },
```

Reply:
Nice call